### PR TITLE
Force use HTTPS when cloning git repos

### PR DESCRIPTION
**Pull request by fridex**
Fixes: https://github.com/fabric8-analytics/fabric8-analytics-worker/issues/182

### CODE REVIEW

File: f8a_worker/utils.py [hunk 1]
```diff
@@ -663,3 +663,15 @@ def decorated(func, *args, **kwargs):
...
+def url2git_repo(url):
+    """Convert URL to git repo URL and force use HTTPS."""
+    if url.startswith('git@'):
+        url = url[len('git@'):]
+        url = url.split(':')
+        if len(url) != 2:
+            raise ValueError("Unable to parse git repo URL '%s'" % str(url))
+        return 'https://{}/{}'.format(url[0], url[1])
+    else:
+        return url
```
Comment (jpopelka) [inline f8a_worker/utils.py]: Could you place it next to `parse_gh_repo()` and add a small unit test, please ?

Comment (fridex) [inline f8a_worker/utils.py]: Sure, will do.

**Comment by fridex** [comment]
[test this]

**Comment by jpopelka** [comment]
I've added similar hack (also due to failing git_stats) recently with #252 so perhaps we should move them to one place. Could you do that ?

**Comment by fridex** [comment]
[test this]

**Comment by fridex** [comment]
[test this]

File: f8a_worker/process.py [hunk 2]
```diff
@@ -58,8 +58,7 @@ def clone(cls, url, path, depth=None, branch=None):
         """
         cls.config()
         # git clone doesn't understand urls starting with: git+ssh, git+http, git+https
-        if url.startswith('git+'):
-            url = url[len('git+'):]
+        url = url2git_repo(url)
```
Comment (jpopelka) [inline f8a_worker/process.py]: You haven't added the `git+` check into `url2git_repo()`

**Comment by fridex** [comment]
@jpopelka ready for review

**Pull request merged**